### PR TITLE
Expandable FAB doc code fixes

### PR DESCRIPTION
**Pull request by EnduringBeta**
_Description of what this PR is changing or adding, and why:_ Updating the code and examples for the [Expandable FAB cookbook entry](https://docs.flutter.dev/cookbook/effects/expandable-fab) to do 2 things:

- The DartPad example has an info-level issue that covers the expanded FAB, so it's actually a bit disruptive.
  - Fixed the issue by removing the `_` before the state class name in all instances
- The action button icons are the same color as the action buttons themselves.
  - Changed the icon color in the action buttons to use `onSecondary` instead of `secondary`, so they show up white on blue

I haven't run any of the code myself except to confirm my changes work in the DartPad. This is my first contribution to this repo, so apologies if I get something wrong.

_Issues fixed by this PR (if any):_ https://github.com/flutter/website/issues/7270

## Presubmit checklist
- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/master/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

File: examples/cookbook/effects/expandable_fab/lib/excerpt1.dart [hunk 1]
```diff
@@ -15,10 +15,10 @@ class ExpandableFab extends StatefulWidget {
   final List<Widget> children;
 
   @override
-  _ExpandableFabState createState() => _ExpandableFabState();
+  ExpandableFabState createState() => ExpandableFabState();
 }
 
-class _ExpandableFabState extends State<ExpandableFab> {
+class ExpandableFabState extends State<ExpandableFab> {
```
Comment (domesticmouse) [inline examples/cookbook/effects/expandable_fab/lib/excerpt1.dart:21]: I can see how you got here, but it's the wrong fix. You are exposing the `State` object as part of the externalised API, when it is a hidden implementation detail by design. Please revert this line back to `class _ExpandableFabState ...`

File: examples/cookbook/effects/expandable_fab/lib/excerpt1.dart [hunk 2]
```diff
@@ -15,10 +15,10 @@ class ExpandableFab extends StatefulWidget {
   final List<Widget> children;
 
   @override
-  _ExpandableFabState createState() => _ExpandableFabState();
+  ExpandableFabState createState() => ExpandableFabState();
```
Comment (domesticmouse) [inline examples/cookbook/effects/expandable_fab/lib/excerpt1.dart:18]: The correct fix for this line is instead:
```dart
State<ExpandableFab> createState() => _ExpandableFabState();
```

**Review by domesticmouse — CHANGES_REQUESTED** [review]

File: examples/cookbook/effects/expandable_fab/lib/excerpt1.dart (hunk 2 above)
Comment (EnduringBeta) [reply examples/cookbook/effects/expandable_fab/lib/excerpt1.dart]: I made the changes you suggested to all Dart files and manually to the MD file. Is there any reading you could point me to on understanding this solution over my original attempt?

**Review by domesticmouse — APPROVED** [review]

File: examples/cookbook/effects/expandable_fab/lib/excerpt1.dart (hunk 2 above)
Comment (domesticmouse) [reply examples/cookbook/effects/expandable_fab/lib/excerpt1.dart]: Books on software API design have been few and far between for the last couple of decades. It's almost as if everyone has gone nuts with network APIs or something. I'd look to some of the classics like Code Complete and Beautiful Code for ideas in this space.

**Comment by domesticmouse** [comment]
Looks like the snippets are out of date, see the bottom of the README on how to update the snippets

**Comment by sfshaza2** [comment]
@EnduringBeta, thanks so much for this PR! Can you refresh the code snippets. From the top of the project:
```
./tool/refresh-code-excerpts.sh
```

**Comment by EnduringBeta** [comment]
Understood, yes. I've been away and not able to get to this. This far I've been editing via GitHub, so I haven't actually grabbed the repo. Hopefully setup will be mildly painless.

**Comment by EnduringBeta** [comment]
I'm on Windows and got this error on `make up` when trying to set up the repo:

```
$ make up
docker-compose up site
time="2022-06-22T17:53:52-04:00" level=warning msg="The \"DISABLE_TESTS\" variable is not set. Defaulting to a blank string."
time="2022-06-22T17:53:52-04:00" level=warning msg="The \"FIREBASE_ALIAS\" variable is not set. Defaulting to a blank string."
time="2022-06-22T17:53:52-04:00" level=warning msg="The \"FIREBASE_TOKEN\" variable is not set. Defaulting to a blank string."
Container website-site-1  Recreate
Container website-site-1  Recreated
Attaching to website-site-1
website-site-1  | bundle exec jekyll serve \
website-site-1  |       --host 0.0.0.0 \
website-site-1  |       --port 4002 \
website-site-1  |       --config _config.yml,_config_dev.yml \
website-site-1  |       --livereload \
website-site-1  |       --incremental \
website-site-1  |       --trace
website-site-1  | Configuration file: /app/_config.yml
website-site-1  | Configuration file: /app/_config_dev.yml
website-site-1  | bundler: failed to load command: jekyll (/usr/local/bundle/bin/jekyll)
website-site-1  | /usr/local/bundle/gems/jekyll-4.2.2/lib/jekyll/external.rb:60:in `require': /app/src/_plugins/breadcrumb.rb:1: unknown regexp option - t (SyntaxError)
website-site-1  | ../../site-shared/src/_plugins/breadcru...
website-site-1  |      ^~~~~
website-site-1  |       from /usr/local/bundle/gems/jekyll-4.2.2/lib/jekyll/external.rb:60:in `block in require_with_graceful_fail'
website-site-1  |       from /usr/local/bundle/gems/jekyll-4.2.2/lib/jekyll/external.rb:57:in `each'
website-site-1  |       from /usr/local/bundle/gems/jekyll-4.2.2/lib/jekyll/external.rb:57:in `require_with_graceful_fail'
website-site-1  |       from /usr/local/bundle/gems/jekyll-4.2.2/lib/jekyll/plugin_manager.rb:89:in `block in require_plugin_files'
website-site-1  |       from /usr/local/bundle/gems/jekyll-4.2.2/lib/jekyll/plugin_manager.rb:87:in `each'
website-site-1  |       from /usr/local/bundle/gems/jekyll-4.2.2/lib/jekyll/plugin_manager.rb:87:in `require_plugin_files'
website-site-1  |       from /usr/local/bundle/gems/jekyll-4.2.2/lib/jekyll/plugin_manager.rb:21:in `conscientious_require'
website-site-1  |       from /usr/local/bundle/gems/jekyll-4.2.2/lib/jekyll/site.rb:131:in `setup'
website-site-1  |       from /usr/local/bundle/gems/jekyll-4.2.2/lib/jekyll/site.rb:36:in `initialize'
website-site-1  |       from /usr/local/bundle/gems/jekyll-4.2.2/lib/jekyll/commands/build.rb:30:in `new'
website-site-1  |       from /usr/local/bundle/gems/jekyll-4.2.2/lib/jekyll/commands/build.rb:30:in `process'
website-site-1  |       from /usr/local/bundle/gems/jekyll-4.2.2/lib/jekyll/command.rb:91:in `block in process_with_graceful_fail'
website-site-1  |       from /usr/local/bundle/gems/jekyll-4.2.2/lib/jekyll/command.rb:91:in `each'
website-site-1  |       from /usr/local/bundle/gems/jekyll-4.2.2/lib/jekyll/command.rb:91:in `process_with_graceful_fail'
website-site-1  |       from /usr/local/bundle/gems/jekyll-4.2.2/lib/jekyll/commands/serve.rb:86:in `block (2 levels) in init_with_program'
website-site-1  |       from /usr/local/bundle/gems/mercenary-0.4.0/lib/mercenary/command.rb:221:in `block in execute'
website-site-1  |       from /usr/local/bundle/gems/mercenary-0.4.0/lib/mercenary/command.rb:221:in `each'
website-site-1  |       from /usr/local/bundle/gems/mercenary-0.4.0/lib/mercenary/command.rb:221:in `execute'
website-site-1  |       from /usr/local/bundle/gems/mercenary-0.4.0/lib/mercenary/program.rb:44:in `go'
website-site-1  |       from /usr/local/bundle/gems/mercenary-0.4.0/lib/mercenary.rb:21:in `program'
website-site-1  |       from /usr/local/bundle/gems/jekyll-4.2.2/exe/jekyll:15:in `<top (required)>'
website-site-1  |       from /usr/local/bundle/bin/jekyll:25:in `load'
website-site-1  |       from /usr/local/bundle/bin/jekyll:25:in `<top (required)>'
website-site-1  |       from /usr/local/bundle/gems/bundler-2.3.16/lib/bundler/cli/exec.rb:58:in `load'
website-site-1  |       from /usr/local/bundle/gems/bundler-2.3.16/lib/bundler/cli/exec.rb:58:in `kernel_load'
website-site-1  |       from /usr/local/bundle/gems/bundler-2.3.16/lib/bundler/cli/exec.rb:23:in `run'
website-site-1  |       from /usr/local/bundle/gems/bundler-2.3.16/lib/bundler/cli.rb:479:in `exec'
website-site-1  |       from /usr/local/bundle/gems/bundler-2.3.16/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
website-site-1  |       from /usr/local/bundle/gems/bundler-2.3.16/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
website-site-1  |       from /usr/local/bundle/gems/bundler-2.3.16/lib/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
website-site-1  |       from /usr/local/bundle/gems/bundler-2.3.16/lib/bundler/cli.rb:31:in `dispatch'
website-site-1  |       from /usr/local/bundle/gems/bundler-2.3.16/lib/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
website-site-1  |       from /usr/local/bundle/gems/bundler-2.3.16/lib/bundler/cli.rb:25:in `start'
website-site-1  |       from /usr/local/bundle/gems/bundler-2.3.16/exe/bundle:48:in `block in <top (required)>'
website-site-1  |       from /usr/local/bundle/gems/bundler-2.3.16/lib/bundler/friendly_errors.rb:117:in `with_friendly_errors'
website-site-1  |       from /usr/local/bundle/gems/bundler-2.3.16/exe/bundle:36:in `<top (required)>'
website-site-1  |       from /usr/local/bundle/bin/bundle:23:in `load'
website-site-1  |       from /usr/local/bundle/bin/bundle:23:in `<main>'
website-site-1  | make: *** [Makefile:64: serve] Error 1
website-site-1 exited with code 2
```

I tried the script you shared, which gave an error because it couldn't find/create `tmp/`. I manually made it, re-ran, and now I get this error:

```
$ ./tool/refresh-code-excerpts.sh
  ╔════════════════════════════════════════════════════════════════════════════╗
  ║ Analytics reporting disabled. In order to enable it, run:                  ║
  ║                                                                            ║
  ║   dart --enable-analytics                                                  ║
  ║                                                                            ║
  ╚════════════════════════════════════════════════════════════════════════════╝

[INFO] Generating build script...
[INFO] Generating build script completed, took 284ms

[INFO] Precompiling build script......
[INFO] Precompiling build script... completed, took 4.5s

[INFO] Initializing inputs
[INFO] Building new asset graph...
[INFO] Building new asset graph completed, took 2.6s

[INFO] Checking for unexpected pre-existing outputs....
[INFO] Checking for unexpected pre-existing outputs. completed, took 1ms

[INFO] Running build...
[INFO] Running build completed, took 650ms

[INFO] Caching finalized dependency graph...
[INFO] Caching finalized dependency graph completed, took 75ms

[SEVERE] No assets exist in C, skipping output
[SEVERE] Unable to create merged directory for /Users/ross/projects/website/tmp/_fragments.
[SEVERE] Failed after 748ms
```

I feel blocked.

**Comment by domesticmouse** [comment]
> I'm on Windows

Our build process isn't tested on, or designed for, running on Windows. It's designed for unix systems. We mainly use macOS laptops, and Linux in the cloud. There are various options for using Linux in the cloud, from GitHub's [Codespaces](https://github.com/features/codespaces) to Google's [Cloud Shell](https://cloud.google.com/shell).

**Comment by domesticmouse** [comment]
Here's the diff of the required change to make CI happy:

```diff
--- a/src/cookbook/effects/expandable-fab.md
+++ b/src/cookbook/effects/expandable-fab.md
@@ -689,7 +689,7 @@ class ActionButton extends StatelessWidget {
       child: IconButton(
         onPressed: onPressed,
         icon: icon,
-        color: theme.colorScheme.secondary,
+        color: theme.colorScheme.onSecondary,
       ),
     );
   }
```

**Comment by EnduringBeta** [comment]
> Here's the diff of the required change to make CI happy:
> 
> ```diff
> --- a/src/cookbook/effects/expandable-fab.md
> +++ b/src/cookbook/effects/expandable-fab.md
> @@ -689,7 +689,7 @@ class ActionButton extends StatelessWidget {
>        child: IconButton(
>          onPressed: onPressed,
>          icon: icon,
> -        color: theme.colorScheme.secondary,
> +        color: theme.colorScheme.onSecondary,
>        ),
>      );
>    }
> ```

Not sure how I missed that, but fixed. I also pulled and pushed latest. Again, disclaimer that I haven't run the scripts.

I'm disappointed that I can't fully contribute via Windows but understand that it's extra work, just like it was extra work for y'all to help get this through. If there's a next time, I'll retry on Mac.

**Pull request merged**